src/main/java/org/broadinstitute/hellbender/utils/pileup/ReadPileup.java

```diff
@@ -0,0 +1,392 @@
+     */
+    public List<GATKRead> getReads() {
+        return pileupElements.stream().map(pe -> pe.getRead()).collect(Collectors.toList());
+    }
+
+    /**
+     * The number of deletion bases in this pileup.
+     */
+    public int getNumberOfDeletions() {
+        //Note: pileups are small so int is fine.
+        return (int)pileupElements.stream().filter(p -> p.isDeletion()).count();
+    }
+
+    /**
+     * The number of reads with mapping quality 0 in this pileup.
+     */
+    public int getNumberOfMappingQualityZeroReads() {
+        //Note: pileups are small so int is fine.
+        return (int)pileupElements.stream().filter(p -> p.getRead().getMappingQuality() == 0).count();
+    }
+
+    /**
+     * The number of reads with deletions after this element.
+     */
+    public int getNumberOfDeletionsAfterThisElement() {
+        //Note: pileups are small so int is fine.
+        return (int)pileupElements.stream().filter(pe -> pe.isBeforeDeletionStart()).count();
+    }
+
+    /**
+     * The number of reads with insertions after this element.
+     */
+    public int getNumberOfInsertionsAfterThisElement() {
+        //Note: pileups are small so int is fine.
+        return (int)pileupElements.stream().filter(pe -> pe.isBeforeInsertion()).count();
+    }
+
+    /**
+     * Returns a list of the offsets in this pileup.
+     * Note: this call costs O(n) and allocates fresh lists each time
+     */
+    public List<Integer> getOffsets() {
+        return pileupElements.stream().map(pe -> pe.getOffset()).collect(Collectors.toList());
+    }
+
+    /**
+     * Returns an array of the bases in this pileup.
+     * Note: this call costs O(n) and allocates fresh array each time
+     */
+    public byte[] getBases() {
+        final byte[] v = new byte[size()];
+        int pos = 0;
+        for (final PileupElement pile : pileupElements) {
+            v[pos++] = pile.getBase();
+        }
+        return v;
+    }
+
+    /**
+     * Returns an array of the quals in this pileup.
+     * Note: this call costs O(n) and allocates fresh array each time
+     */
+    public byte[] getQuals() {
+        final byte[] v = new byte[size()];
+        int pos = 0;
+        for (final PileupElement pile : pileupElements) {
+            v[pos++] = pile.getQual();
+        }
+        return v;
+    }
+
+    /**
+     * Get an array of the mapping qualities.
+     */
+    public int[] getMappingQuals() {
+        final int[] v = new int[size()];
+        int pos = 0;
+        for (final PileupElement pile : pileupElements) {
+            v[pos++] = pile.getRead().getMappingQuality();
+        }
+        return v;
+    }
+
+    private static String quals2String(final byte[] quals) {
+        final StringBuilder qualStr = new StringBuilder();
+        for (int qual : quals) {
+            qual = Math.min(qual, 63);              // todo: fixme, this isn't a good idea
```

Comment:
If it's not a good idea, let's not do it. Can this sketchy method be refactored away completely?